### PR TITLE
unique species aggregation

### PR DESCRIPTION
**Pull request by prakhar-s**
added my code for getting unique species aggregation for stats view

### CODE REVIEW

File: src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java [hunk 1]
```diff
@@ -30,17 +33,20 @@
...
 import com.strandls.observation.es.util.ObservationListPageMapper;
 import com.strandls.observation.pojo.AllRecoSugguestions;
 import com.strandls.observation.pojo.MapAggregationResponse;
+import com.strandls.observation.pojo.MapAggregationStatsResponse;
 import com.strandls.observation.pojo.ObservationHomePage;
 import com.strandls.observation.pojo.ObservationListData;
 import com.strandls.observation.pojo.RecoIbp;
 import com.strandls.observation.pojo.RecoShow;
+import com.strandls.observation.pojo.Recommendation;
 import com.strandls.observation.service.ObservationListService;
```
Comment (akrudra) [inline src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java:51]: seems good check for unused imports and if any remove them

**Review by akrudra — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -445,7 +444,8 @@
 					<executable>java</executable>
 					<mainClass>org.openapitools.codegen.OpenAPIGenerator</mainClass>
 					<commandlineArgs>
-						generate -i ${project.basedir}/target/swagger.json -g java -o target/sdk --skip-validate-spec -c swagger-config.json
+						generate -i ${project.basedir}/target/swagger.json
```
[the remark this answers is not in the document]
Comment (prakhar-s) [reply pom.xml]: fixed that command in single line as requested

File: src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java [hunk 3]
```diff
@@ -72,14 +85,24 @@ public ObservationListData getObservationList(String index, String type, MapSear
...
 				geoHashResult = geoHashAggregationData.getGeoHashData();
 				totalCount = geoHashAggregationData.getTotalCount();
 
-			} else {
+			}
+
+			else if (view.equalsIgnoreCase("stats")) {
+
+				statsAggregates = aggregationStatsResult;
+				// System.out.println(aggregationStatsResult.getGroupUniqueSpecies());
```
[the remark this answers is not in the document]
Comment (prakhar-s) [reply src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java]: removed this sop statement

File: src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java (hunk 1 above)
Comment (prakhar-s) [reply src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java]: removed unused imports and other unused statements

**Pull request closed without merging**